openedx/features/course_duration_limits/access.py

```diff
@@ -109,7 +109,7 @@ def get_user_course_expiration_date(user, course):
         # Content availability date is equivalent to max(enrollment date, course start date)
         # for most people. Using the schedule date will provide flexibility to deal with
         # more complex business rules in the future.
-        content_availability_date = enrollment.schedule.start
+        content_availability_date = enrollment.schedule.start_date
         # We have anecdotally observed a case where the schedule.start was
         # equal to the course start, but should have been equal to the enrollment start
         # https://openedx.atlassian.net/browse/PROD-58
```

Comment:
I'm also seeing `schedule.start` still being used in the test code below - shouldn't all these references be changed as well?:
```

/Users/jeskew/docker_devstack/edx-platform/openedx/core/djangoapps/schedules/tests/test_resolvers.py:
  121              async_send_task=Mock(name='async_send_task'),
  122              site=self.site_config.site,
  123:             target_datetime=enrollment.schedule.start,
  124              day_offset=-7,
  125              bin_num=CourseUpdateResolver.bin_num_for_user_id(self.user.id),

/Users/jeskew/docker_devstack/edx-platform/openedx/core/djangoapps/schedules/tests/test_signals.py:
  156          course = _create_course_run(self_paced=True, start_day_offset=5)  # course starts in future
  157          enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT)
  158:         assert _strip_secs(enrollment.schedule.start) == _strip_secs(course.start)
  159  
  160      @override_waffle_flag(CREATE_SCHEDULE_WAFFLE_FLAG, True)
  ...
  168          course = _create_course_run(self_paced=True, start_day_offset=-5)  # course already started
  169          enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT)
  170:         assert _strip_secs(enrollment.schedule.start) == _strip_secs(enrollment.created)
  171  
  172  
```

Reply:
I do tend to agree with Julia, if we are now only using start_date, our tests should be writing and reading from start_date.

Reply:
abdul, can both of those test changes be also moved from the old PR to this PR?

Reply:
you mean from https://github.com/edx/edx-platform/pull/22392 to current PR?

Reply:
Also, just wanted to acknowledge that it is getting very late for you.  I can take over this PR if you will be stopping work for today.

Reply:
> you mean from #22392 to current PR?

yes